src/instruction.rs

```diff
@@ -1180,4 +1193,51 @@ RX(%a) 0",
 
         assert_eq!(expected_program, program);
     }
+
+    #[test]
+    fn it_parses_memory_reference_from_str() {
+        [
+            "_",
+            "a",
+            "a---b",
+            "_a_b_",
+            "a-2_b-2",
+            "_[0]",
+            "a[1]",
+            "a---b[2]",
+            "_a_b_[3]",
+            "a-2_b-2[4]",
+        ]
+        .iter()
+        .for_each(|&s| {
+            assert!(
+                MemoryReference::from_str(s).is_ok(),
```

Comment:
Can you change this to assert that the parsed value is as expected, not just that it didn't error?